qa/L0_infer/test.sh

```diff
@@ -168,23 +168,33 @@ for TARGET in cpu gpu; do
 
     set +e
 
-    # python unittest seems to swallow ImportError and still return 0
-    # exit code. So need to explicitly check CLIENT_LOG to make sure
-    # we see some running tests
     python $INFER_TEST >$CLIENT_LOG 2>&1
     if [ $? -ne 0 ]; then
         cat $CLIENT_LOG
         echo -e "\n***\n*** Test Failed\n***"
         RET=1
     fi
 
-    grep -c "HTTPSocketPoolResponse status=200" $CLIENT_LOG
+    # At the end of $CLIENT_LOG there is a single line JSON containing the
+    # result of unittests.
+    test_result_json=`tail -n 1 $CLIENT_LOG`
+
+    echo $test_result_json | jq
     if [ $? -ne 0 ]; then
         cat $CLIENT_LOG
-        echo -e "\n***\n*** Test Failed To Run\n***"
+        echo -e "\n***\n*** Test Failed\n***"
         RET=1
+    else
+        num_failures=`echo $test_result_json | jq .failures`
+        num_tests=`echo $test_result_json | jq .total`
+        if [ $num_failures != "0" ] || [ $num_tests -le 0 ]; then
```

Comment:
$num_tests should be compared against the exact number of tests that are expected so that we can detect if fewer were run for some reason. I would add an EXPECTED_NUM_TESTS at the start of the file which the expected value. When someone changes the tests they will need to also update that.

Reply:
Done.